groot/routes.py

```diff
@@ -1,17 +1,52 @@
 from flask import Flask, Response, render_template, request, redirect, url_for, flash
 from run import app, db
-
+from groot.models import *
+import hashlib
 
 
 @app.route("/")
 @app.route('/login', methods=['POST', 'GET'])
 def login():
-    return render_template('login.html')
-        
+    if request.method == 'POST':
+        try:
+            email = request.form.get('email')
+            password = request.form.get('password')    
+            user = User.query.filter_by(email=email).first()
+            if user:
+                hash_pw = hashlib.sha1(password.encode('utf-8')).hexdigest().upper()
+                if user.password == hash_pw:
+                    return redirect(url_for('dashboard'))  
```

Comment:
This is ok, but maybe try to redirect the user to the "next page" instead (in case his session has timed out, he'll be able to return to the same/desired page and not just the dashboard)

Reply:
I'm on it!